### PR TITLE
[rpm] Integrate 1.41.0 nghttp2. JB#50106

### PR DESCRIPTION
**Pull request by rainemak**
This needs to mirror still.

### CODE REVIEW

**Comment by rainemak** [comment]
requires:
https://github.com/sailfishos/c-ares/pull/1 &&
https://github.com/sailfishos/libev/pull/1

**Comment by rainemak** [comment]
Dropped WIP and updated git submodule link.

File: rpm/nghttp2.spec [hunk 1]
```diff
@@ -0,0 +1,106 @@
+Summary: Experimental HTTP/2 client, server and proxy
+Name: nghttp2
+Version: 1.41.0
+Release: 1
+License: MIT
+URL: https://nghttp2.org/
+Source0:    %{name}-%{version}.tar.bz2
+
+BuildRequires: c-ares-devel
```
Comment (rainemak) [inline rpm/nghttp2.spec]: Let's swap these to pkgconfig() for each which have pc file.

Comment (okodron) [reply rpm/nghttp2.spec]: Agree.

File: rpm/nghttp2.spec [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+BuildRequires: gcc-c++
```
Comment (okodron) [inline rpm/nghttp2.spec:9]: Such explicit dependency could be dropped, I guess.

File: rpm/nghttp2.spec [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+%prep
+%autosetup -p1 -n %{name}-%{version}/%{name}
+
+# make fetch-ocsp-response use Python 3
+sed -e '1 s|^#!/.*python|&3|' -i script/fetch-ocsp-response
+
+%build
+autoreconf --force --install
+# From configure: enable-lib-only is a short hand for --disable-app --disable-examples --disable-hpack-tools --disable-python-bindings
+%configure PYTHON=%{__python3}              \
```
Comment (okodron) [inline rpm/nghttp2.spec:57]: You can use `%reconfigure` instead of `autoreconf` & `%configure`.

File: rpm/nghttp2.spec [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+
+%post -p /sbin/ldconfig -n libnghttp2
+%postun -p /sbin/ldconfig -n libnghttp2
+
+%check
+# test the just built library instead of the system one, without using rpath
+export "LD_LIBRARY_PATH=$RPM_BUILD_ROOT%{_libdir}:$LD_LIBRARY_PATH"
+make check
+
+%files -n libnghttp2
```
Comment (okodron) [inline rpm/nghttp2.spec:90]: Better to add `%defattr(-,root,root,-)`

**Comment by rainemak** [comment]
Dropped also systemd dependency as we're not building at the moment any nghttp2 servers.

**Comment by rainemak** [comment]
If you're pondering about package structure it is the same what fedora has but fedora has also some ng binaries or servers. This approach we'd be nicely aligned with that and would be simple to add those binaries to the package if we need them.

**Comment by okodron** [comment]
> If you're pondering about package structure it is the same what fedora has but fedora has also some ng binaries or servers. This approach we'd be nicely aligned with that and would be simple to add those binaries to the package if we need them.

Sure, my suggestions are more cosmetic.

File: rpm/nghttp2.spec (hunk 4 above)
Comment (rainemak) [reply rpm/nghttp2.spec:90]: Added

File: rpm/nghttp2.spec (hunk 2 above)
Comment (rainemak) [reply rpm/nghttp2.spec:9]: Dropped as well

**Review by mkosola — APPROVED** [review]

**Review by okodron — APPROVED** [review]

**Pull request merged**